server/handler/base.go

```diff
@@ -391,6 +402,73 @@ func selectionToReviewersRequest(s reviewer.Selection) github.ReviewersRequest {
 	return req
 }
 
+func (b *Base) dedupDiscardedReviews(discardedReviews []*common.DiscardedReview) []*common.DiscardedReview {
+	var reviews []*common.DiscardedReview
+
+Reviews:
+	for _, d := range discardedReviews {
+		for _, r := range reviews {
+			if r.ID == d.ID {
+				continue Reviews
+			}
+		}
+		reviews = append(reviews, d)
+	}
+
+	return reviews
+}
+
+func (b *Base) findResultsWithDiscardedReviews(result *common.Result) []*common.Result {
+	var results []*common.Result
+	for _, c := range result.Children {
+		results = append(results, b.findResultsWithDiscardedReviews(c)...)
+	}
+	if len(result.Children) == 0 && len(result.DiscardedReviews) > 0 && result.Error == nil {
+		results = append(results, result)
+	}
+	return results
+}
+
+func (b *Base) dismissStaleReviewsForResult(ctx context.Context, v4client *githubv4.Client, result common.Result) error {
+	var reviews []*common.DiscardedReview
+
+	results := b.findResultsWithDiscardedReviews(&result)
+	for _, r := range results {
+		dedupedReviews := b.dedupDiscardedReviews(r.DiscardedReviews)
```

Comment:
I think we need to deduplicate the list _after_ combining the reviews from all of the results. I expect duplicates to appear because multiple rules dismiss the same review rather than a single result dismissing the same review twice.

Reply:
i removed the dedup, since I changed it to use the cached reviews from prctx instead.